File: dotnet/test/Azure.Iot.Operations.Protocol.MetlTests/TestCaseActionReceiveResponse.cs
namespace Azure.Iot.Operations.Protocol.MetlTests
{
    public class TestCaseActionReceiveResponse : TestCaseAction
    {
        public static string? DefaultTopic;
        public static string? DefaultPayload;
        public static string? DefaultContentType;
        public static int? DefaultFormatIndicator;
        public static int? DefaultCorrelationIndex;
        public static int? DefaultQos;
        public static TestCaseDuration? DefaultMessageExpiry;
        public static string? DefaultStatus;
        public static string? DefaultStatusMessage;
        public static string? DefaultIsApplicationError;
        public static string? DefaultInvalidPropertyName;
        public static string? DefaultInvalidPropertyValue;

        public string? Topic { get; set; } = DefaultTopic;

        public string? Payload { get; set; } = DefaultPayload;

        public bool BypassSerialization { get; set; }

        public string? ContentType { get; set; } = DefaultContentType;

        public int? FormatIndicator { get; set; } = DefaultFormatIndicator;

        public Dictionary<string, string> Metadata { get; set; } = new();

        public int? CorrelationIndex { get; set; } = DefaultCorrelationIndex;

        public int? Qos { get; set; } = DefaultQos;

        public TestCaseDuration? MessageExpiry { get; set; } = DefaultMessageExpiry;

        public string? Status { get; set; } = DefaultStatus;

        public string? StatusMessage { get; set; } = DefaultStatusMessage;

        public string? IsApplicationError { get; set; } = DefaultIsApplicationError;

        public string? InvalidPropertyName { get; set; } = DefaultInvalidPropertyName;

        public string? InvalidPropertyValue { get; set; } = DefaultInvalidPropertyValue;

        public int? PacketIndex { get; set; }
    }
}

File: go/test/protocol/test_case_action_receive_response.go
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
package protocol

import (
	"gopkg.in/yaml.v3"
)

type testCaseActionReceiveResponse struct {
	Topic                *string           `yaml:"topic"`
	Payload              *string           `yaml:"payload"`
	ContentType          *string           `yaml:"content-type"`
	FormatIndicator      *int              `yaml:"format-indicator"`
	Metadata             map[string]string `yaml:"metadata"`
	CorrelationIndex     *int              `yaml:"correlation-index"`
	Qos                  *int              `yaml:"qos"`
	MessageExpiry        *TestCaseDuration `yaml:"message-expiry"`
	Status               *string           `yaml:"status"`
	StatusMessage        *string           `yaml:"status-message"`
	IsApplicationError   *string           `yaml:"is-application-error"`
	InvalidPropertyName  *string           `yaml:"invalid-property-name"`
	InvalidPropertyValue *string           `yaml:"invalid-property-value"`
	PacketIndex          *int              `yaml:"packet-index"`
}

type TestCaseActionReceiveResponse struct {
	testCaseActionReceiveResponse
}

func (receiveResponse *TestCaseActionReceiveResponse) UnmarshalYAML(
	node *yaml.Node,
) error {
	*receiveResponse = TestCaseActionReceiveResponse{}

	receiveResponse.Topic = TestCaseDefaultInfo.Actions.ReceiveResponse.GetTopic()
	receiveResponse.Payload = TestCaseDefaultInfo.Actions.ReceiveResponse.GetPayload()
	receiveResponse.ContentType = TestCaseDefaultInfo.Actions.ReceiveResponse.GetContentType()
	receiveResponse.FormatIndicator = TestCaseDefaultInfo.Actions.ReceiveResponse.GetFormatIndicator()
	receiveResponse.CorrelationIndex = TestCaseDefaultInfo.Actions.ReceiveResponse.GetCorrelationIndex()
	receiveResponse.Qos = TestCaseDefaultInfo.Actions.ReceiveResponse.GetQos()
	receiveResponse.Status = TestCaseDefaultInfo.Actions.ReceiveResponse.GetStatus()
	receiveResponse.StatusMessage = TestCaseDefaultInfo.Actions.ReceiveResponse.GetStatusMessage()
	receiveResponse.IsApplicationError = TestCaseDefaultInfo.Actions.ReceiveResponse.GetIsApplicationError()
	receiveResponse.InvalidPropertyName = TestCaseDefaultInfo.Actions.ReceiveResponse.GetInvalidPropertyName()
	receiveResponse.InvalidPropertyValue = TestCaseDefaultInfo.Actions.ReceiveResponse.GetInvalidPropertyValue()

	defaultMessageExpiry := TestCaseDefaultInfo.Actions.ReceiveResponse.GetMessageExpiry()
	receiveResponse.MessageExpiry = &defaultMessageExpiry

	return node.Decode(&receiveResponse.testCaseActionReceiveResponse)
}
